Dockerfile_test

```diff
@@ -0,0 +1,16 @@
+FROM kudobuilder/golang:1.12
+
+WORKDIR $GOPATH/src/github.com/kudobuilder/kudo
+COPY .git-credentials .git-credentials
+
+
+COPY pkg/    pkg/
+COPY config/    config/
+COPY cmd/    cmd/
+COPY go.mod go.mod
+COPY go.sum go.sum
+COPY Makefile Makefile
+
+RUN cp .git-credentials ~/.git-credentials
+
+RUN make test
```

Comment:
So the tests get run during the docker build and I suppose the build step fails if the RUN command fails? Do the error message(s) get exposed to circle ci?

Reply:
yes, see this as example of failed run https://circleci.com/gh/kudobuilder/kudo/626

Reply:
No strong opinion, I was just surprised to see the run test as part of the docker build. 
In particular about debugging etc. E.g., if we build a container and then run test inside the container we exec get into the container or even run tests in parallel.
As mentioned mostly interested in understanding the pros/cons of both approaches

Reply:
thought about it and like it - I think it's more docker-way of doing things. Thanks!